docs/src/pages/components/data-grid/filtering/CustomRatingOperator.js

```diff
@@ -40,7 +40,7 @@ RatingInputValue.propTypes = {
   applyValue: PropTypes.func.isRequired,
   item: PropTypes.shape({
     columnField: PropTypes.string,
-    id: PropTypes.number,
+    id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
```

Comment:
A side question. Why does the items have an `id`? (What problem does it solve?)

Reply:
we need it for keys in the list, and we also use them to compare filters, delete them...

Reply:
> for keys in the list

Why not use the index position?

> use them to compare filters, delete them

Why not compare the JS object reference?